azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/SubscriptionScoped.java

```diff
@@ -44,4 +48,22 @@ public T subscriptions(@Nonnull final List<Subscription> subscriptions) {
     public T subscription(@Nonnull final Subscription subscription) {
         return this.subscriptions(Collections.singletonList(subscription));
     }
+
+    public T subscription(@Nonnull final String subscriptionId) {
+        return this.subscriptions(subscriptionId);
+    }
+
+    public T subscriptions(@Nonnull final String... subscriptions) {
+        assert ArrayUtils.isNotEmpty(subscriptions) : "subscriptions can not be empty!";
+        final List<Subscription> subscriptionList = Azure.az(IAzureAccount.class).account().getSubscriptions().stream()
+                .filter(subscription -> ArrayUtils.contains(subscriptions, subscription.getId()))
+                .collect(Collectors.toList());
+        return this.subscriptions(subscriptionList);
+    }
+
+    // If only one subscription is selected, using it as default subscription
+    public Subscription getDefaultSubscription() {
+        return Optional.of(CollectionUtils.size(subscriptions) == 1 ? subscriptions.get(0) : null)
```

Comment:
`subscriptions` is null if user doesn't explicitly scope subscriptions (defaultly scoped to selected subscriptions)